### PR TITLE
Update dublicate object lookup process for FDM 6.4+

### PR DESCRIPTION
**Pull request by 119Vik**
According to "Full-Text Search" feature been introduced in the FDM v 6.4 ansible codebase was updated to optimize ansible modules for v6.4 and all newer versions of FDM.

### CODE REVIEW

File: module_utils/configuration.py [hunk 1]
```diff
@@ -287,13 +292,23 @@ def match_filters(filter_params, obj):
...
         )
         return (i for i in item_generator if match_filters(filters, i))
 
+    def _stringify_name_filter(self, filters):
+        if self._system_info['version'] >= '6.4.0':
+            return "fts~%s" % filters['name']
+        return "name:%s" % filters['name']
+
+    def _fetch_system_info(self):
+        params = {"path_params": {"objId": "default"}}
```
Comment (annikulin) [inline module_utils/configuration.py]: There is a `PATH_PARAMS_FOR_DEFAULT_OBJ` variable in `configuration.py` file. Please, reuse it.

File: module_utils/configuration.py [hunk 2]
```diff
@@ -215,6 +215,11 @@ def __init__(self, conn, check_mode=False):
         self._check_mode = check_mode
         self._operation_checker = OperationChecker
 
+        system_info = self._fetch_system_info()
```
Comment (annikulin) [inline module_utils/configuration.py]: Let's introduce some lazy loading for the system info as it's not needed for all `add`/`delete`/`get` operations and it requires an extra request on each task. E.g., let's create a `get_build_version` method that will be called from `get_objects_by_filter` and fetch system info if needed.

Comment (119Vik) [reply module_utils/configuration.py]: make sense. fixed.

File: module_utils/configuration.py (hunk 1 above)
Comment (119Vik) [reply module_utils/configuration.py]: good catch.

File: module_utils/configuration.py [hunk 3]
```diff
@@ -287,13 +287,28 @@ def match_filters(filter_params, obj):
...
 
+    def _stringify_name_filter(self, filters):
+        build_version = self.get_build_version()
+        if build_version >= '6.4.0':
+            return "fts~%s" % filters['name']
+        return "name:%s" % filters['name']
+
+    def _fetch_system_info(self):
+        params = {ParamName.PATH_PARAMS: PATH_PARAMS_FOR_DEFAULT_OBJ}
+        resp = self.send_general_request('getSystemInformation', params)
```
Comment (annikulin) [inline module_utils/configuration.py]: Let's cache `getSystemInformation` response so we don't do extra requests if `get_build_version` is called multiple times.

Comment (119Vik) [reply module_utils/configuration.py]: solved

**Pull request merged**